spec/Classes.tex

```diff
@@ -663,73 +600,453 @@ \subsection{User-Defined Constructors}
+\end{chapel}
+\begin{chapelpost}
+writeln(p);
+\end{chapelpost}
+\begin{chapeloutput}
+x = 1.0
+{x = 2.0, y = 2.0}
+\end{chapeloutput}
+The first statement in the initializer initializes field \chpl{x} to the value
+of the formal \chpl{x}. The second statement simply reads the value of the
+initialized field. The third statement reads the value of the field, doubles
+it, and {\em assigns} the result to the field \chpl{x}.
+\end{chapelexample}
+
+If a field is used before it is initialized, an compile-time error will be
+issued.
+
+\begin{chapelexample}{fieldUsedBeforeInitialized.chpl}
+In the following code:
+\begin{chapel}
+class Point {
+  var x, y : real;
+
+  proc init(x: real, y: real) {
+    writeln(this.x); // Error: use of uninitialized field!
+    this.x = x;
+    this.y = y;
+    writeln(this.y);
+  }
+}
+var p = new Point(1.0, 2.0);
+\end{chapel}
+\begin{chapelcompopts}
+--no-warnings
+\end{chapelcompopts}
+
+The first statement in the initializer reads the value of uninitialized field
+\chpl{x}, so the compiler will issue an error.
+
+\begin{chapelprintoutput}
+fieldUsedBeforeInitialized.chpl:5: error: field "x" used before it is initialized
+\end{chapelprintoutput}
+
+\end{chapelexample}
+
+\subsubsection{Omitting Field Initializations}
+\label{Omitting_Field_Initializations}
+\index{classes!initializers!user-defined!omitting}
+
+Some classes have many fields, some of which might have default values. Instead
+of requiring the user to initialize each field explicitly, the language allows
+field initializations to be omitted if the field has a type or if the field
+has an initialization expression. The compiler will insert initialization
+statements for such fields based on their types and default values.
+
+\begin{chapelexample}{fieldInitOmitted.chpl}
+In the following code:
+\begin{chapel}
+class LabeledPoint {
+  var x : real;
+  var y : real;
+  var msg : string = 'Unlabeled';
+
+  proc init(x: real, y: real) {
+    this.x = x;
+    this.y = y;
+    // compiler inserts "this.msg = 'Unlabeled'";
+  }
+
+  proc init(msg : string) {
+    // compiler inserts "this.x = 0.0;"
+    // compiler inserts "this.y = 0.0;"
+    this.msg = msg;
+  }
+}
+
+var A = new LabeledPoint(2.0, 3.0);
+var B = new LabeledPoint("Origin");
+\end{chapel}
+\begin{chapelpost}
+writeln(A);
+writeln(B);
+\end{chapelpost}
+\begin{chapelcompopts}
+--no-warnings
+\end{chapelcompopts}
+\begin{chapeloutput}
+{x = 2.0, y = 3.0, msg = Unlabeled}
+{x = 0.0, y = 0.0, msg = Origin}
+\end{chapeloutput}
+The first initializer initializes the values of fields \chpl{x} and \chpl{y},
+and the compiler inserts initialization for the \chpl{msg} field by using
+its default value. The second initializer initializes the \chpl{msg} field,
+and the compiler inserts initialization for fields \chpl{x} and \chpl{y}
+based on the type of those fields (\rsec{Default_Values_For_Types}).
+\end{chapelexample}
+
+Unchecked, this language feature could lead to confusing behavior for fields
```

Comment:
This seems a bit weird to me - "editorializing" or something. Maybe this and the following paragraph could be combined. If that's not enough, you can use a `\chpl{rationale}` section.